onadata/libs/serializers/data_serializer.py

```diff
@@ -110,6 +111,27 @@ def to_representation(self, instance):
         ret = super(DataInstanceSerializer, self).to_representation(instance)
         if 'json' in ret:
             ret = ret['json']
+        return ret
+
+
+class TableauDataSerializer(serializers.ModelSerializer):
+    """
+    TableauDataSerializer class - cleans out instance fields.
+    """
+    json = JsonField()
+
+    class Meta:
+        model = Instance
+        fields = ('json', )
+
+    def to_representation(self, instance):
+        ret = super(TableauDataSerializer, self).to_representation(instance)
+        if 'json' in ret:
+            ret = ret['json']
+            # Remove metadata fields from the instance
+            for field in METADATA_FIELDS:
```

Comment:
this can call the function you've defined, but to avoid circular imports you'll have to move that function to a utils namespace

Reply:
Much much cleaner. Appreciate this.